src/smt/optimization_solver.h

```diff
@@ -72,38 +93,58 @@ enum class OptResult
  * - whether it's maximize/minimize
  * - and whether it's signed for BitVectors
  */
-class Objective
+class OptimizationObjective
 {
  public:
+  /**
+   * An enum for optimization queries.
+   * Represents whether an objective should be minimized or maximized
+   */
+  enum ObjectiveType
+  {
+    MINIMIZE,
+    MAXIMIZE,
+  };
+
   /**
    * Constructor
-   * @param n the optimization target node
+   * @param target the optimization target node
    * @param type speficies whether it's maximize/minimize
    * @param bvSigned specifies whether it's using signed or unsigned comparison
    *    for BitVectors this parameter is only valid when the type of target node
    *    is BitVector
    **/
-  Objective(Node n, ObjectiveType type, bool bvSigned = false);
-  ~Objective(){};
+  OptimizationObjective(Node target, ObjectiveType type, bool bvSigned = false)
```

Comment:
These short function are inlined according to this: https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#inline-functions
They are short enough to justify the inline.